contracts/implementation/LeveragedPool.sol

```diff
@@ -402,22 +410,38 @@ contract LeveragedPool is ILeveragedPool, Initializable {
      * @notice Unpauses the pool
      * @dev Prevents all state updates until unpaused
      */
-    function unpause() external onlyGov {
+    function unpause() external override onlyGov {
         paused = false;
         emit Unpaused();
     }
 
+    /**
+     * @dev Check invariants before function body only. This is used in functions where the state of the pool is updated after exiting PoolCommitter (i.e. executeCommitments)
+     */
+    modifier checkInvariantsAndUnpausedBeforeOnly() {
```

Comment:
naming sounds confusing to me. 
perfer this naming:
checkInvariantsBeforeAction
checkInvariantsAfterAction

or

checkInvariantsBeforeFunction
checkInvariantsAfterFunction